### PR TITLE
[RFC] Yarn Workspaces

### PR DESCRIPTION
**Pull request by thejameskyle**
[[Rendered](https://github.com/thejameskyle/rfcs-1/blob/workspaces/accepted/0000-workspaces.md)]

### CODE REVIEW

**Comment by bestander** [comment]
Hey @thejameskyle, that is a nice document you've had hidden from us :)

We've been considering trade offs and implementation details for months and decided to go with the phased approach. It's too large for one person to claim ownership over the full feature and we invite you to participate in the phased approach.

I have set up phases in this issue https://github.com/yarnpkg/yarn/issues/3294.
If you like you can claim a **piece** of it from RFC to full implementation.

**Comment by thejameskyle** [comment]
I'm aware of that, I was in those meetings early on 😛 

There's nothing about this that cannot be broken down into phases, but the feature should really be considered all at once.

Also, I shared the private gist with you and @cpojer months ago

**Comment by torifat** [comment]
> Right now we use a question field to manually type in a version. However, we should change that to a multi-choice selector (See Inquirer.js).

Recently I [moved](/yarnpkg/yarn/pull/3207) `Inquirer` inside our `ConsoleReporter`. I was thinking about refactoring the question/select. It will improve DX of several workflows.

File: accepted/0000-workspaces.md [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+pieces of tools into their own packages because it would make development
+harder.
+
+If Yarn had a way of developing many packages as a single project which removed
+the additional maintenance cost of being able to test changes across packages,
+it would encourage more tools to abstract core functionality out.
+
+# Detailed design
+
+This is the bulk of the RFC. Explain the design in enough detail for somebody
```
Comment (JulianLeviston) [inline accepted/0000-workspaces.md]: Does this section need to be deleted, or filled in? Looks like boilerplate.

**Comment by bestander** [comment]
Hey, @thejameskyle, thanks for sending this RFC, very inspiring.

Currently we think that we can address specifics faster in a phased approach that I track in https://github.com/yarnpkg/yarn/issues/3294.

Having focused discussions around particular isolated features allows us to have a faster turnaround and also experiment with code as the discussions happen.

Your spec has a good coverage around publishing experience and you have plenty related experience from lerna, do you want to champion that part of workspaces from RFC to implementation?

**Comment by cpojer** [comment]
Closing this PR as it was superseded by the RFCs for Workspaces phase 1, 2 and 3.

**Pull request closed without merging**